openedxscorm/scormxblock.py

```diff
@@ -331,6 +335,35 @@ def publish_grade(self):
         self.runtime.publish(
             self, "grade", {"value": self.get_grade(), "max_value": self.weight},
         )
+        self.publish_completion()
+
+
+    def publish_completion(self):
+        """
+        Mark scorm xbloxk as completed if user has completed the scorm course unit.
+
+        it will work along with the edX completion tool: https://github.com/edx/completion
+        """
+        if not completion_waffle.waffle().is_enabled(completion_waffle.ENABLE_COMPLETION_TRACKING):
+            return
+
+        if XBlockCompletionMode.get_mode(self) != XBlockCompletionMode.COMPLETABLE:
+            return
+
+        completion_value = 0.0
+        if not self.has_score:
+            # componenet does not have any score
+            if self.get_completion_status() == 'completed':
+                completion_value = 1.0
+        else:
+            if self.get_completion_status() in ['passed', 'failed']:
+                completion_value = 1.0
+
+        data = {
+            'completion': completion_value
+        }
+        self.runtime.publish(self, "completion", data)
```

Comment:
```suggestion
        self.runtime.publish(self, 'completion', data)
```

Reply:
for consistency, I am using double quotes.